lib/evmone/instructions_calls.cpp

```diff
@@ -79,9 +85,13 @@ Result call_impl(StackTop stack, int64_t gas_left, ExecutionState& state) noexce
 
     evmc_message msg{.kind = to_call_kind(Op)};
     msg.flags = (Op == OP_STATICCALL) ? uint32_t{EVMC_STATIC} : state.msg->flags;
+    if (dst != code_addr)
+        msg.flags |= EVMC_DELEGATED;
+    else
+        msg.flags &= ~std::underlying_type_t<evmc_flags>{EVMC_DELEGATED};
```

Comment:
This hack point to that maybe `msg.flags` is not the best way to signal delegation (because it is not propagated unlike static mode).

Alternative could be a separate delegation address field in `evmc_message`, or another flag field.